test/api/test_module_construct.cpp

```diff
@@ -1,16 +1,56 @@
+#include <numeric>
 #include <migraphx/migraphx.h>
 #include <migraphx/migraphx.hpp>
 #include "test.hpp"
 
+void run_test_on_gpu(const migraphx::program p)
+{
+    // make a copy of the program
+    migraphx_program_t p_ref_c;
+    migraphx_program_create(&p_ref_c);
+    migraphx_program_assign_to(p_ref_c, p.get_handle_ptr());
+    auto p_ref = migraphx::program(p_ref_c, migraphx::own{});
+    // compile on GPU
+    migraphx::compile_options options;
+    options.set_offload_copy();
+    p.compile(migraphx::target("gpu"), options);
+    // set parameters_map
+    migraphx::program_parameters pp;
+    auto param_shapes = p.get_parameter_shapes();
+    for(auto&& name : param_shapes.names())
+    {
+        pp.add(name, migraphx::argument::generate(param_shapes[name]));
+    }
+    // get results from GPU
+    auto outputs  = p.eval(pp);
+    auto output   = outputs[0];
+    auto lens     = output.get_shape().lengths();
+    auto elem_num = std::accumulate(lens.begin(), lens.end(), 1, std::multiplies<std::size_t>());
+    float* output_data_ptr = reinterpret_cast<float*>(output.data());
+    std::vector<float> output_vec(output_data_ptr, output_data_ptr + elem_num);
+    // compile using ref target
+    p_ref.compile(migraphx::target("ref"));
+    // get results from Ref target
+    auto outputs_ref = p_ref.eval(pp);
+    auto output_ref  = outputs_ref[0];
+    lens             = output_ref.get_shape().lengths();
+    elem_num         = std::accumulate(lens.begin(), lens.end(), 1, std::multiplies<std::size_t>());
+    float* output_ref_data_ptr = reinterpret_cast<float*>(output_ref.data());
+    std::vector<float> output_ref_vec(output_ref_data_ptr, output_ref_data_ptr + elem_num);
+    // compare the results;
+    CHECK(output_vec == output_ref_vec);
```

Comment:
There is no need to put the argument into a vector, you can use equality on the `argument` directly.

Reply:
Did equality for `add_op` test.  But for the `if_op`, test i am still converting into a vector. For some reason the return argument has type of `bool` .  but the data of the return argument matches to the expected. Not able to figure out where this type mismatch is coming from.

Reply:
there is type mismatch because i am using lambda which requires copy and assignment.